pkg/compiler/codegen.go

```diff
@@ -653,11 +676,42 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 
 		return nil
 
+	case *ast.BranchStmt:
+		var label string
+		if n.Label != nil {
+			label = n.Label.Name
+		} else if n.Tok == token.BREAK {
+			label = c.labelSwitch
+		} else if n.Tok == token.CONTINUE {
+			label = c.labelFor
+		}
+
+		switch n.Tok {
+		case token.BREAK:
+			end := c.getLabelByName(getLabelEnd(label))
+			emit.Jmp(c.prog.BinWriter, opcode.JMP, int16(end))
```

Comment:
Side note: I'd probably use `uint16` here (and in other cases like this) and check for `uint16` overflow in `newLabel` (and for `int16` overflow in `writeJumps`).

Reply:
Easy #688 .